module3/src/MyList.java

```diff
@@ -0,0 +1,697 @@
+     *
+     * <p>Lists that support this operation may place limitations on what
+     * elements may be added to this list.  In particular, some
+     * lists will refuse to add null elements, and others will impose
+     * restrictions on the type of elements that may be added.  List
+     * classes should clearly specify in their documentation any restrictions
+     * on what elements may be added.
+     *
+     * @param e element to be appended to this list
+     * @return <tt>true</tt> (as specified by {@link Collection#add})
+     * @throws UnsupportedOperationException if the <tt>add</tt> operation
+     *         is not supported by this list
+     * @throws ClassCastException if the class of the specified element
+     *         prevents it from being added to this list
+     * @throws NullPointerException if the specified element is null and this
+     *         list does not permit null elements
+     * @throws IllegalArgumentException if some property of this element
+     *         prevents it from being added to this list
+     */
+    @Override
+    public boolean add(T e) {
+        if (size <= capacity - GROW_FACTOR) {
+            capacity += DEFAULT_CAPACITY;
+            Object[] result = new Object[capacity];
+            for (int i = 0; i < size; i++) {
+                result[i] = this.elementData[i];
+            }
+            this.elementData = result;
+        }
+        this.elementData[size] = e;
+        size++;
+        return true;
+    }
+
+    /**
+     * Removes the first occurrence of the specified element from this list,
+     * if it is present (optional operation).  If this list does not contain
+     * the element, it is unchanged.  More formally, removes the element with
+     * the lowest index <tt>i</tt> such that
+     * <tt>(o==null&nbsp;?&nbsp;get(i)==null&nbsp;:&nbsp;o.equals(get(i)))</tt>
+     * (if such an element exists).  Returns <tt>true</tt> if this list
+     * contained the specified element (or equivalently, if this list changed
+     * as a result of the call).
+     *
+     * @param o element to be removed from this list, if present
+     * @return <tt>true</tt> if this list contained the specified element
+     * @throws ClassCastException if the type of the specified element
+     *         is incompatible with this list
+     * (<a href="Collection.html#optional-restrictions">optional</a>)
+     * @throws NullPointerException if the specified element is null and this
+     *         list does not permit null elements
+     * (<a href="Collection.html#optional-restrictions">optional</a>)
+     * @throws UnsupportedOperationException if the <tt>remove</tt> operation
+     *         is not supported by this list
+     */
+    @Override
+    public boolean remove(Object o) {
+        if (this.indexOf(o) > -1) {
+            Object result = remove(this.indexOf(o));
+            if (o == null) {
+                return true;
+            }
+            if (result != null) {
```

Comment:
you can get rid of this `if`, and just `return result != null`

Reply:
I agree

Reply:
seems like it still not changed